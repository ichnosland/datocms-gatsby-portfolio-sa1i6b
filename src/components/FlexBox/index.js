/**
*
* FlexBox
*
*/
import styled, { css } from 'styled-components';

import Div from '../Div';
import media from 'style/mediainjector';

export const FlexBox = styled(Div)`
  display: flex;
  flex-direction: ${(props) => props.direction ? props.direction : 'row'};
  justify-content: ${(props) => props.justifyContent ? props.justifyContent : 'center'};
  align-items: ${(props) => props.alignItems ? props.alignItems : 'center'};
  align-content: ${(props) => props.alignContent ? props.alignContent : 'center'};
  flex-wrap: ${(props) => props.wrap ? props.wrap : 'nowrap'};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const FlexChild = styled(FlexBox)`
  order: ${(props) => props.order};
  flex-grow: ${(props) => props.flexGrow};
  flex-basis: ${(props) => props.flexBasis};
  align-self: ${(props) => props.alignSelf};
`;

export const FlexWrap = styled(FlexBox)`
  position: relative;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100%;
  height: 100%;
  ${media.desktop`
    &:before {
      content: '';
      display: block;
    }
  `}
  ${media.mobile`
    display: block;
  `}
`;

export default FlexBox;
