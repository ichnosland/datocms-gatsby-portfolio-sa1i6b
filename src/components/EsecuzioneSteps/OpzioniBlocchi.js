/**
*
* OpzioniBlocchi
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import FlexBox, { FlexChild } from 'components/FlexBox';
import Div from 'components/Div';
import media from 'style/mediainjector';
import { colore } from 'style/color';

const dragColor = [
  '#f5ee97',
  '#d6f4a0',
  '#E0D5E3',
  '#ffd8bf',
  '#cde5ff',
];

const TagOpzioniBlocchi = styled(Div)`
  padding: 10px 10px 20px;
  ${media.lt667`
    padding-bottom: 10px;
  `}
`;

export const WrapOpzioniBlocchi = styled(FlexBox)`
  position: fixed;
  width: 90%;
  height: 90%;
  left: 5%;
  top: 5%;
  flex-direction: column;
  padding: 20px;
  background-color: ${colore.ui.contrast};
  border-radius: ${(props) => props.theme.radius.general};
  ${media.lt667`
    padding: 10px;
  `}
`;

WrapOpzioniBlocchi.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export const OpzioneBlocchi = styled(FlexChild)`
  position: relative;
  width: 100%;
  margin-bottom: 20px;
  background-color: ${(props) => props.dragColor ? props.dragColor : 'rgba(0, 0, 0, 0.1)'};
  border-radius: ${(props) => props.theme.radius.general};
  flex: 1 auto;
  flex-grow: 1;
  &:last-child {
    margin-bottom: 0;
  }
  ${media.lt667`
    margin-bottom: 10px;
  `}
`;

OpzioneBlocchi.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

class OpzioniBlocchi extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { titolo, opzioni, selectFunction, indiceBlocco } = this.props;

    return (
      <WrapOpzioniBlocchi>
        <TagOpzioniBlocchi>
          {titolo}
        </TagOpzioniBlocchi>
        {opzioni.map((opzione) => (
          <OpzioneBlocchi
            key={`blocco_${indiceBlocco}_${opzione.indiceEtichetta}`}
            dragColor={dragColor[opzione.indiceEtichetta - 1]}
            onClick={() => selectFunction({
              [indiceBlocco]: {
                labelOpzione: opzione.etichetta,
                indiceLabel: opzione.indiceEtichetta,
              },
            })}
          >
            {opzione.etichettaComposita}
          </OpzioneBlocchi>
        ))}
      </WrapOpzioniBlocchi>
    );
  }
}

OpzioniBlocchi.propTypes = {
  titolo: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array
  ]).isRequired,
  opzioni: PropTypes.arrayOf(
    PropTypes.shape({
      etichetta: PropTypes.string.isRequired,
      etichettaComposita: PropTypes.oneOfType([
        PropTypes.array,
        PropTypes.string,
      ]).isRequired,
      indiceEtichetta: PropTypes.number.isRequired,
    }),
  ).isRequired,
  selectFunction: PropTypes.func.isRequired,
  indiceBlocco: PropTypes.number.isRequired,
};

export default OpzioniBlocchi;
