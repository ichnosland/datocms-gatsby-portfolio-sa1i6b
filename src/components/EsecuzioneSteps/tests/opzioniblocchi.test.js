import React from 'react';
import { shallow } from 'enzyme';

import FlexBox, { FlexChild } from '../FlexBox';
import OpzioniBlocchi, { WrapOpzioniBlocchi, OpzioneBlocchi } from '../OpzioniBlocchi';

const titolo = 'Titolo';

const theme = {
  radius: {
    general: '80px',
  },
};

const opzioni = [{
  id: 1,
  indiceEtichetta: 1,
  etichetta: '',
  titolo: 'titolo',
  etichettaComposita: '',
}];

const indiceBlocco = 3;

const selectFunction = () => { };

describe('<OpzioniBlocchi /> ', () => {
  it('should render an <div>', () => {
    const renderedComponent = shallow(
      <OpzioniBlocchi
        titolo={titolo}
        opzioni={opzioni}
        indiceBlocco={indiceBlocco}
        selectFunction={selectFunction}
      />
    );
    expect(renderedComponent.find(WrapOpzioniBlocchi).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should work with latex', () => {
    const mockProps = {
      titolo: [<span key="k">titolo</span>],
      opzioni: [{
        etichetta: 'etichetta 1',
        etichettaComposita: [<span key="k">etichetta 1</span>],
        indiceEtichetta: 1,
      }, {
        etichetta: 'etichetta 2',
        etichettaComposita: [<span key="l">etichetta 1</span>],
        indiceEtichetta: 2,
      }],
      selectFunction: jest.fn(),
      indiceBlocco: 123,
    };

    const renderedComponent = shallow(
      <OpzioniBlocchi {...mockProps} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('il click dell\'opzione deve chiamare la selectFunction', () => {
    const mockProps = {
      titolo: 'titolo',
      opzioni: [{
        etichetta: 'etichetta 1',
        etichettaComposita: 'etichetta 1',
        indiceEtichetta: 1,
      }, {
        etichetta: 'etichetta 2',
        etichettaComposita: 'etichetta 2',
        indiceEtichetta: 2,
      }],
      selectFunction: jest.fn(),
      indiceBlocco: 123,
    };

    const renderedComponent = shallow(
      <OpzioniBlocchi {...mockProps} />
    );

    expect(mockProps.selectFunction).not.toHaveBeenCalled();
    renderedComponent.find(OpzioneBlocchi).at(0).simulate('click');
    expect(mockProps.selectFunction).toHaveBeenCalledWith({
      123: {
        indiceLabel: 1,
        labelOpzione: 'etichetta 1',
      },
    });
  });
});

describe('<WrapOpzioniBlocchi /> ', () => {
  it('should render a FlexBox', () => {
    const renderedComponent = shallow(<WrapOpzioniBlocchi />);
    expect(renderedComponent.find(FlexBox).length).toEqual(1);
  });

  it('should render its props w/ theme', () => {
    const renderedComponent = shallow(<WrapOpzioniBlocchi theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<OpzioneBlocchi /> ', () => {
  it('should render an FlexChild', () => {
    const renderedComponent = shallow(<OpzioneBlocchi />);
    expect(renderedComponent.find(FlexChild).length).toEqual(1);
  });

  it('should render its props w/ theme', () => {
    const renderedComponent = shallow(<OpzioneBlocchi theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dragColor', () => {
    const dragColor = 'red';
    const renderedComponent = shallow(<OpzioneBlocchi theme={theme} dragColor={dragColor} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
