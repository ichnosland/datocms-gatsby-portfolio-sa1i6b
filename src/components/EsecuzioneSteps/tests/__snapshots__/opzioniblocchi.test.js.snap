// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OpzioneBlocchi />  should render its css w/ props dragColor 1`] = `
.c0 {
  position: relative;
  width: 100%;
  margin-bottom: 20px;
  background-color: red;
  border-radius: 80px;
  -webkit-flex: 1 auto;
  -ms-flex: 1 auto;
  flex: 1 auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0:last-child {
  margin-bottom: 0;
}

@media (max-width:41.6875em) {
  .c0 {
    margin-bottom: 10px;
  }
}

<FlexBox__FlexChild
  className="c0"
  dragColor="red"
  theme={
    Object {
      "radius": Object {
        "general": "80px",
      },
    }
  }
/>
`;

exports[`<OpzioneBlocchi />  should render its props w/ theme 1`] = `
.c0 {
  position: relative;
  width: 100%;
  margin-bottom: 20px;
  background-color: rgba(0,0,0,0.1);
  border-radius: 80px;
  -webkit-flex: 1 auto;
  -ms-flex: 1 auto;
  flex: 1 auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0:last-child {
  margin-bottom: 0;
}

@media (max-width:41.6875em) {
  .c0 {
    margin-bottom: 10px;
  }
}

<FlexBox__FlexChild
  className="c0"
  theme={
    Object {
      "radius": Object {
        "general": "80px",
      },
    }
  }
/>
`;

exports[`<OpzioniBlocchi />  should render an <div> 1`] = `
<OpzioniBlocchi__WrapOpzioniBlocchi
  theme={
    Object {
      "radius": Object {
        "general": "6px",
      },
    }
  }
>
  <OpzioniBlocchi__TagOpzioniBlocchi>
    Titolo
  </OpzioniBlocchi__TagOpzioniBlocchi>
  <OpzioniBlocchi__OpzioneBlocchi
    dragColor="#f5ee97"
    key="blocco_3_1"
    onClick={[Function]}
    theme={
      Object {
        "radius": Object {
          "general": "6px",
        },
      }
    }
  />
</OpzioniBlocchi__WrapOpzioniBlocchi>
`;

exports[`<OpzioniBlocchi />  should work with latex 1`] = `
<OpzioniBlocchi__WrapOpzioniBlocchi
  theme={
    Object {
      "radius": Object {
        "general": "6px",
      },
    }
  }
>
  <OpzioniBlocchi__TagOpzioniBlocchi>
    <span
      key="k"
    >
      titolo
    </span>
  </OpzioniBlocchi__TagOpzioniBlocchi>
  <OpzioniBlocchi__OpzioneBlocchi
    dragColor="#f5ee97"
    key="blocco_123_1"
    onClick={[Function]}
    theme={
      Object {
        "radius": Object {
          "general": "6px",
        },
      }
    }
  >
    <span
      key="k"
    >
      etichetta 1
    </span>
  </OpzioniBlocchi__OpzioneBlocchi>
  <OpzioniBlocchi__OpzioneBlocchi
    dragColor="#d6f4a0"
    key="blocco_123_2"
    onClick={[Function]}
    theme={
      Object {
        "radius": Object {
          "general": "6px",
        },
      }
    }
  >
    <span
      key="l"
    >
      etichetta 1
    </span>
  </OpzioniBlocchi__OpzioneBlocchi>
</OpzioniBlocchi__WrapOpzioniBlocchi>
`;

exports[`<WrapOpzioniBlocchi />  should render its props w/ theme 1`] = `
.c0 {
  position: fixed;
  width: 90%;
  height: 90%;
  left: 5%;
  top: 5%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  background-color: rgb(255,255,255);
  border-radius: 80px;
}

@media (max-width:41.6875em) {
  .c0 {
    padding: 10px;
  }
}

<FlexBox
  className="c0"
  theme={
    Object {
      "radius": Object {
        "general": "80px",
      },
    }
  }
/>
`;
